### PR TITLE
Adds FETCH_POST_STATUS action

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR adds `FETCH_POST_STATUS` action which we will utilize to fix https://github.com/wordpress-mobile/WordPress-Android/issues/10951. The action is implemented for both REST and XML-RPC sites and both of them have connected tests.

@malinajirka Would you like me to open a separate PR for combining the errors in a single method in `PostXMLRPCClient`? I only introduced the method to be shared between `fetchPost` and `fetchPostStatus` but then realized we are doing the exact same error handling in every action, so I combined them all. I am happy to separate that into a new PR if you feel it'll help.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/post/PostXMLRPCClient.java [hunk 1]
```diff
@@ -653,4 +627,29 @@ private static PostModel postResponseObjectToPostModel(@NonNull Map postObject,
         }
         return params;
     }
+
+    private List<Object> fetchPostParams(final PostModel post, final SiteModel site) {
```
Comment (malinajirka) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/post/PostXMLRPCClient.java]: We might want to consider renaming the method so it doesn't start with "fetch" which evokes to me that the method will hit the server, wdyt? Perhaps, something like "createFetchPostParams"?

Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/post/PostXMLRPCClient.java]: Good catch! I was testing things quickly and looks like I forgot to do this while polishing.

Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/post/PostXMLRPCClient.java]: Fixed in 5e8656278021357d7d97bfe30c4b57205e11430a.

**Comment by oguzkocer** [comment]
@malinajirka I think we should just merge this PR. It doesn't contain any breaking changes, it has its own connected tests and most importantly I think we might try to get the comments getting disabled PR into the frozen branch next week and having the FluxC side already merged would simplify things for us.

What do you think?

**Pull request merged**